plugins/out_stackdriver/stackdriver_conf.c

```diff
@@ -414,68 +455,61 @@ struct flb_stackdriver *flb_stackdriver_conf_create(struct flb_output_instance *
         }
     }
 
-    if (flb_sds_cmp(ctx->resource, "k8s_container",
-                    flb_sds_len(ctx->resource)) == 0 ||
-        flb_sds_cmp(ctx->resource, "k8s_node",
-                    flb_sds_len(ctx->resource)) == 0 ||
-        flb_sds_cmp(ctx->resource, "k8s_pod",
-                    flb_sds_len(ctx->resource)) == 0) {
+    set_resource_type(ctx);
 
-        ctx->is_k8s_resource_type = FLB_TRUE;
+    if (resource_api_has_required_labels(ctx) == FLB_FALSE) {
```

Comment:
Why do we need to validate all other labels below after we checked resource_api_has_required_labels? Could we combine them into one function?